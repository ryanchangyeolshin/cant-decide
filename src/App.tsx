import React, { useState } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import DecisionProvider from './context/DecisionProvider';
import NavMenuBar from './components/NavMenuBar';
import SideNavMenu from "./components/SideNavMenu";
import ContentContainer from './components/ContentContainer';
import WinningChoiceContainer from './components/WinningChoiceContainer';
import './App.css';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    app: {
      background: 'linear-gradient(45deg, #F4ECDA 30%, #FDFBF7 90%)',
      height: "100%"
    }
  }),
);

const App: React.FC = () => {
  const classes = useStyles();
  const [sideMenu, setSideMenu] = useState<boolean>(false);

  const toggleDrawer = (open: boolean) => (
    event: React.KeyboardEvent | React.MouseEvent,
  ) => {
    if (
      event.type === 'keydown' &&
      ((event as React.KeyboardEvent).key === 'Tab' ||
        (event as React.KeyboardEvent).key === 'Shift')
    ) {
      return;
    }

    setSideMenu(open);
  };

  return (
    <div className={classes.app}>
      <DecisionProvider>
        <NavMenuBar toggleDrawer={toggleDrawer} />
        <SideNavMenu sideMenu={sideMenu} toggleDrawer={toggleDrawer} />
        <Router>
          <Switch>
            <Route path="/decision">
              <WinningChoiceContainer />
            </Route>
            <Route path="/">
              <ContentContainer />
            </Route>
          </Switch>
        </Router>
      </DecisionProvider>
    </div>
  );
}

export default App;
