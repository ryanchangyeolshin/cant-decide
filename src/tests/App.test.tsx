import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

describe("This suit is to test the App component", () => {
  test('Snapshot of App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot()
  });
});
