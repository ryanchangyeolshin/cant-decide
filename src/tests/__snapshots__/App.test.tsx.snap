// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the App component Snapshot of App 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-app-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-3 MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <span
          class="makeStyles-typed-8"
        >
          <span />
          <span
            class="typed-cursor"
          >
            |
          </span>
        </span>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text login-button MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div />
    <div
      class="makeStyles-containerWrapper-12"
    >
      <div
        class="MuiTypography-root makeStyles-contentContainer-13 MuiTypography-body1"
      >
        <div
          class="makeStyles-inputContainer-15"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-choiceInput-16 MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-18 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="standard-basic"
              id="standard-basic-label"
            >
              Choice
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-17"
                id="standard-basic"
                placeholder="Please provide your choice here..."
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-20 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-22"
                >
                  <span>
                    Choice
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-randomize-19 MuiButton-containedPrimary Mui-disabled Mui-disabled"
            data-testid="randomize-button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Randomize
            </span>
          </button>
        </div>
        <div
          class="makeStyles-list-14"
        >
          <nav
            aria-label="main mailbox folders"
            class="MuiList-root MuiList-padding"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
